kratos/tests/test_function_parser_utility.py

```diff
@@ -127,6 +127,48 @@ def test_ApplyFunctionToNodesUtility(self):
         for node in model_part.Nodes:
             self.assertEqual(node.GetSolutionStepValue(KM.VISCOSITY) - (node.Y + 2.0 * node.X), 0.0)
 
+    def test_ApplyFunctionToNodesUtilityTimeEvolutionTernary(self):
+        parameters = KM.Parameters ("""{}""")
+        function = KM.GenericFunctionUtility("1.5*(0.5*(1-cos(0.5*pi*t))*2.0)*(4.0/0.1681)*y*(0.41-y) if t<2.0 else 1.5*(2.0)*(4.0/0.1681)*y*(0.41-y)", parameters)
+
+        this_model = KM.Model()
+        model_part = this_model.CreateModelPart("Main", 2)
+        current_process_info = model_part.ProcessInfo
+        current_process_info[KM.DOMAIN_SIZE] = 2
+
+        model_part.AddNodalSolutionStepVariable(KM.DISPLACEMENT)
+        model_part.AddNodalSolutionStepVariable(KM.VISCOSITY)
+        model_part.AddNodalSolutionStepVariable(KM.VELOCITY)
+
+        model_part_io = KM.ModelPartIO(GetFilePath("auxiliar_files_for_python_unittest/mdpa_files/test_model_part_io_read"))
+        model_part_io.ReadModelPart(model_part)
+
+        current_process_info[KM.TIME] = 0.0
+        time = current_process_info[KM.TIME]
+        while time < 3.0:
+            current_process_info[KM.TIME] = current_process_info[KM.TIME] + 1.0
+            time = current_process_info[KM.TIME]
+
+            utility = KM.ApplyFunctionToNodesUtility(model_part.Nodes, function)
+            utility.ApplyFunction(KM.VISCOSITY, time)
+
+            if time < 2.0:
+                for node in model_part.Nodes:
+                    self.assertEqual(node.GetSolutionStepValue(KM.VISCOSITY) - (1.5*(0.5*(1-math.cos(0.5*math.pi*time))*2.0)*(4.0/0.1681)*node.Y*(0.41-node.Y)), 0.0)
+            else:
+                for node in model_part.Nodes:
+                    self.assertAlmostEqual(node.GetSolutionStepValue(KM.VISCOSITY), 1.5*(2.0)*(4.0/0.1681)*node.Y*(0.41-node.Y))
+
+    @KratosUnittest.expectedFailure
```

Comment:
please check for the actual error with `assertRaisesRegex`

Reply:
Should be fine now

Reply:
```suggestion
```